### DISCUSSION TITLE
Add support for more granular key events (e.g. key down)

### DISCUSSION THREAD

**Discussion by IsaacLambat** [qtile/qtile#3068] [Ideas]
It would be great if Qtile had support for more granular key events rather than just key press. For example, key down and key up which means that keys can be binded to do different actions while holding a key down. This could further be extended for the mouse binds as well. 

A use case this might be helpful for is for when the mod key is held, focus moves with the mouse (by setting the `follow_mouse_focus` to true on key down, and false on key up). This is useful for moving floating windows, as they do not need to be clicked first to get focus then moved. 

The only current way to do this that I can think of would be to use an external application for key binds and then [piping in the commands](https://docs.qtile.org/en/latest/manual/commands/scripting.html). Would be nice to have this supported natively.

**Comment by m-col** [comment]
Agreed! There are two PRs in the works that will provide more configurability to bindings:

 - https://github.com/qtile/qtile/pull/2925 which makes it possible to specify whether or not a key event is swallowed
 - https://github.com/qtile/qtile/pull/2785 which will enable key-release bindings

If you'd be interested to give them a test, getting feedback and confirmation that they are working as expected is very helpful :)
